modules/kubernetes-addons/crossplane/data.tf

```diff
@@ -0,0 +1,40 @@
+data "aws_caller_identity" "current" {}
+
+data "aws_partition" "current" {}
+
+data "kubectl_path_documents" "aws_provider" {
+  pattern = "${path.module}/aws-provider/provider-aws.yaml"
+  vars = {
+    provider-aws-version = var.crossplane_provider_aws.provider_aws_version
+    iam-role-arn         = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/${var.eks_cluster_id}-provider-aws--irsa"
```

Comment:
remove one hypen at the end (ie `${var.eks_cluster_id}-provider-aws-irsa`)

Reply:
We need hyphen here as i am replacing  only`*` from this `provider-aws-*`  in IRSA module to avoid naming issue